controllers/api/v1/base.go

```diff
@@ -134,6 +134,28 @@ func (c *baseAPIController) serveJSON(data interface{}) {
 	}
 }
 
+func (c *baseAPIController) serveListJSON(data interface{}, meta *jsonapi.Meta, links *jsonapi.Links) {
+	response, err := jsonapi.Marshal(data)
+	if err != nil {
+		c.renderError(err, http.StatusInternalServerError)
+	}
+
+	payload, ok := response.(*jsonapi.ManyPayload)
+	if !ok {
+		c.renderError(err, http.StatusInternalServerError)
+	}
+
+	payload.Meta = meta
+	payload.Links = links
+
+	c.Data["json"] = payload
```

Comment:
@junan I think you don't feel like to merge this block with the previous function, right? 😎

```golang
c.Data["json"] = payload
err = c.ServeJSON()
if err != nil {
    c.renderError(err, http.StatusInternalServerError)
}
```